CallAutomation_OutboundCalling/CallAutomation_OutboundCalling/Program.cs

```diff
@@ -45,55 +72,68 @@
 
+            var choices = GetChoices();
+
             // prepare recognize tones
-            CallMediaRecognizeDtmfOptions callMediaRecognizeDtmfOptions = new CallMediaRecognizeDtmfOptions(new PhoneNumberIdentifier(targetPhonenumber), maxTonesToCollect: 1);
-            callMediaRecognizeDtmfOptions.Prompt = new FileSource(new Uri(callbackUriHost + "/audio/MainMenu.wav"));
-            callMediaRecognizeDtmfOptions.InterruptPrompt = true;
-            callMediaRecognizeDtmfOptions.InitialSilenceTimeout = TimeSpan.FromSeconds(5);
+            var recognizeOptions = GetMediaRecognizeChoiceOptions(MainMenu, targetPhonenumber, choices);
 
             // Send request to recognize tones
-            await callMedia.StartRecognizingAsync(callMediaRecognizeDtmfOptions);
+            await callMedia.StartRecognizingAsync(recognizeOptions);
         }
         if (parsedEvent is RecognizeCompleted recognizeCompleted)
         {
-            // Play audio once recognition is completed sucessfully
-            string selectedTone = ((DtmfResult)recognizeCompleted.RecognizeResult).ConvertToString();
-
-            switch (selectedTone)
-            {
-                case "1":
-                    await callMedia.PlayToAllAsync(new FileSource(new Uri(callbackUriHost + "/audio/Confirmed.wav")));
-                    break;
-
-                case "2":
-                    await callMedia.PlayToAllAsync(new FileSource(new Uri(callbackUriHost + "/audio/Goodbye.wav")));         
-                    break;
-
-                default:
-                    //invalid tone
-                    await callMedia.PlayToAllAsync(new FileSource(new Uri(callbackUriHost + "/audio/Invalid.wav")));
-                    break;
-            }
+            var choiceResult = recognizeCompleted.RecognizeResult as ChoiceResult;
+            var labelDetected = choiceResult?.Label;
+            var phraseDetected = choiceResult?.RecognizedPhrase;
+            // If choice is detected by phrase, choiceResult.RecognizedPhrase will have the phrase detected, 
+            // If choice is detected using dtmf tone, phrase will be null 
+            logger.LogInformation("Recognize completed succesfully, labelDetected={labelDetected}, phraseDetected={phraseDetected}", labelDetected, phraseDetected);
+            var textToPlay = labelDetected.Equals(ConfirmChoiceLabel, StringComparison.OrdinalIgnoreCase) ? ConfirmedText : CancelText;
+
+            await HandlePlayAsync(callMedia, textToPlay);
         }
         if (parsedEvent is RecognizeFailed recognizeFailedEvent)
         {
-            logger.LogInformation($"RecognizeFailed parsedEvent received for call connection id: {parsedEvent.CallConnectionId}");
-
-            // Check for time out, and then play audio message
-            if (recognizeFailedEvent.ReasonCode.Equals(MediaEventReasonCode.RecognizeInitialSilenceTimedOut))
+            var context = recognizeFailedEvent.OperationContext;
```

Comment:
there is a better way of doing this in C#,  if(parsedEvent is RecognizedFailed recognizeFailedEvent {OperationContext : "somecontext})